feat: Allow experimental editor opt out

dcalhoun opened the pull request:
## Description
<!-- Describe the changes, why they are needed, and how they address the issue -->
Utilize a remote feature flag to enable a phased roll out. Allow opting out of
the experimental editor for a TBD period of time.

Because WP-Android's experimental features are separate from the remote 
feature flag architecture, we add a "Disable [feature]" toggle to use in tandem 
with the remote feature flag. 

The enable or disable toggle is conditionally presented based upon the remote
feature flag status. 

> [!Note]
> Given WP-Android's experimental features are detached from remote feature,
> an edge case is possible where a user would be unable to opt into 
> GutenbergKit. It feels too unlikely and unimportant to address. 
>
> Achieving the edge case would require:
> 1. Include the user in a remote, phased roll out cohort. 
> 1. User toggles the "Disable experimental block editor" ON. 
> 1. Remove user from the aforementioned cohort. 
> 1. A user enabling "Experimental block editor" would have no impact at this point. 

Ref CMM-291.

## Testing instructions
<!-- Consider listing specific testing steps for helping reviewers evaluate the efficacy of the changes -->
> [!Tip]
> An example feature flag can be found in 181762-ghe-Automattic/wpcom. 

**1: When remote feature flag is not truthy, app presents enable toggle**
1. Navigate to _Me_ → _Experimental Features_. 
2. Verify the _Experimental block editor_ toggle is available, controlling GBK as expected. 

**2: When remote feature flag is truthy, app presents disable toggle**
1. Set a truthy `gutenberg_kit` feature flag in your WPCOM sandbox. 
2. Route your test device `public-api.wordpress.com` traffic to your sandbox. 
1. Navigate to _Me_ → _Experimental Features_. 
3. Verify the _Disable experimental block editor_ toggle is available, controlling GBK as expected. 

| Light | Dark |
| - | - |
| ![light-mode](https://github.com/user-attachments/assets/255e8e38-a374-4653-b048-298cd13fcb4f) | ![dark-mode](https://github.com/user-attachments/assets/a10f60e2-7d74-4570-87ce-81b83ab48b2a) |

<!--

Consider testing the following *yourself* before requesting a review:

- Compatibility with WordPress.com sites and self-hosted Jetpack sites
- Both portrait and landscape orientations
- Light and dark modes
- Dynamic font sizes: larger, smaller and bold text
- High contrast mode
- Screen reader experience (e.g., Talkback)
- Languages with large words or with letters/accents not frequently used in English
- Right-to-left languages layout
- Multi-tasking: split view and slide over

-->

dcalhoun committed:
feat: Allow opting out of experimental editor

Leverage remote feature flag for a phased roll out, but allow opting out
for a TBD period of time.

dcalhoun committed:
feat: Experimental features display description

Mitigate confusion regarding the impact of features.

dcalhoun committed:
feat: Note the temporary availability of disabling GutenbergKit

As we perform a phased roll out, user may disable GutenbergKit for a TBD
period of time. It is worth communicating the ability to disable will
eventually be removed.

dcalhoun committed:
feat: Add info icon above experimental features note

Improve vertical rhythm and clearly separate sections.

dangermattic commented:
<!--
  0 Errors
  1 Warning: <code>strings.xml</code> files...
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><code>strings.xml</code> files should only be updated on release branches, when the translations are downloaded by our automation.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

dcalhoun committed:
feat: Toggle experimental feature flag based on remote flag status

If the remote flag is enabled, we want to provide the user an opt-out
toggle. Otherwise, we present the ability to enable the experimental
editor.

dcalhoun committed:
style: Address lint warnings

jkmassel reviewed (approved):
This worked as described after following the testing instructions
